### PR TITLE
Update documentation on mixins/intercept

### PR DESCRIPTION
**Pull request by peol**
Clarifies the mixin/intercept APIs.

See it in action here: https://github.com/qlik-oss/enigma.js/blob/update-docs/docs/api.md#api-documentation

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -96,11 +124,36 @@ Mixins that are bound to several different types can find the current API type i
...
+
+### `mixin.init(args)`
+
+This function will be executed for each generated API.
+
+See below what `args` contains.
+
+| Property                | Type     | Description |
+|-------------------------|----------|-------------|
+| `config`                | Object   | A reference to the normalized enigma.js configuration object. |
```
Comment (marcusoffesson) [inline docs/api.md]: Skip the word _normalize_. IMHO, it confuses more than it clarifies.

File: src/session.js [hunk 2]
```diff
@@ -182,6 +182,7 @@ class Session {
     });
     const response = this.rpc.send(data);
     request.id = data.id;
+    request.retry = () => this.send(request);
```
Comment (marcusoffesson) [inline src/session.js]: Nice!

File: docs/api.md [hunk 3]
```diff
@@ -96,11 +124,36 @@ Mixins that are bound to several different types can find the current API type i
 or `type` members. [`this.type`](#apitype) would for instance return `GenericObject` and
 [`this.genericType`](#apigenerictype) would return `barchart`.
 
-See the [Mixins examples](/examples/README.md#mixins) on how to use it.
+See the [Mixins examples](/examples/README.md#mixins) on how to use it, below is an outline of what the mixin
```
Comment (marcusoffesson) [inline docs/api.md]: How about adding a small section on the order of which mixins are applied. Or is that obvious?

Comment (peol) [inline docs/api.md]: No, it's probably a good idea. I'll add it!

File: docs/api.md (hunk 1 above)
Comment (peol) [inline docs/api.md]: I might reformulate this, I just want to clarify that it's a configuration containing e.g. `Promise` and others that we default to if they are not passed in.

**Pull request merged**